### PR TITLE
Reuse up to `max_examples` from database when `Phase.generate` is disabled

### LINKED ISSUE #2585: Reuse up to `max_examples` examples when generation is disabled
https://github.com/HypothesisWorks/hypothesis/blob/caccabbd107757ee6e9de1de7e3cff5474d9ccc5/hypothesis-python/src/hypothesis/internal/conjecture/engine.py#L544

No matter how many examples are in the database, we'll use at most 10% of our example budget replaying them - even if the generation phase is disabled.  It would be nice if we could replay up to `max_examples` when `Phase.generate not in self.settings.phases`.

### PR DESCRIPTION
**Pull request by oxfordhalfblood**
Fixes #2585.

### CODE REVIEW

File: hypothesis-python/src/hypothesis/internal/conjecture/engine.py [hunk 1]
```diff
@@ -542,6 +542,7 @@ def reuse_existing_examples(self):
                 self.settings.database.fetch(self.database_key), key=sort_key
             )
             desired_size = max(2, ceil(0.1 * self.settings.max_examples))
+            desired_size = (desired_size*0.1 if Phase.generate in self.settings.phases else desired_size)
```
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/internal/conjecture/engine.py:545]:
```suggestion
            factor = 0.1 if (Phase.generate in self.settings.phases) else 1
            desired_size = max(2, ceil(factor * self.settings.max_examples))
```

**Comment by Zac-HD** [comment]
We'll also need to add release notes, in a new file `hypothesis-python/RELEASE.rst`:

```rst
RELEASE_TYPE: patch

...
```

File: AUTHORS.rst [hunk 2]
```diff
@@ -126,3 +126,4 @@ their individual contributions.
 * `Wilfred Hughes <https://www.github.com/wilfred>`_
 * `Zac Hatfield-Dodds <https://www.github.com/Zac-HD>`_ (zac.hatfield.dodds@gmail.com)
 * `Zebulun Arendsee <https://www.github.com/arendsee>`_ (zbwrnz@gmail.com)
+* `Afrida Tabassum <https://github.com/oxfordhalfblood>`_ (afrida@gmail.com)
```
Comment (Zac-HD) [inline AUTHORS.rst:129]: This list is alphabetised, so you should probably be closer to the top :wink:

Comment (oxfordhalfblood) [reply AUTHORS.rst:129]: Awesome!

File: hypothesis-python/RELEASE.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+RELEASE_TYPE: patch
+
+This patch ensures that, when the ``generate`` :obj:`~hypothesis.settings.phase` is disabled,
```
Comment (Zac-HD) [inline hypothesis-python/RELEASE.rst:3]:
```suggestion
This patch ensures that, when the ``generate`` :obj:`~hypothesis.settings.phases` is disabled,
```

:hand_over_mouth: whoops... this typo was my fault :disappointed_relieved:

Comment (oxfordhalfblood) [reply hypothesis-python/RELEASE.rst]: cdf51943dceb27452900c1e5ded97369a4d4d1d2 Haha should have cross checked with the codebase.

**Review by Zac-HD — APPROVED** [review]
I'm happy with this, but we need approval specifically from @DRMacIver to merge changes to the `conjecture` internals :slightly_smiling_face:

**Review by DRMacIver — APPROVED** [review]

**Pull request merged**